integrations-service/integrations/utils/integrations/mailgun.py

```diff
@@ -0,0 +1,63 @@
+import aiohttp
+from beartype import beartype
+from tenacity import retry, stop_after_attempt, wait_exponential
+
+from ...autogen.Tools import MailgunSendEmailArguments, MailgunSetup
+from ...env import mailgun_api_key  # Import env to access environment variables
+from ...models import MailgunSendEmailOutput
+
+
+@beartype
+@retry(
+    wait=wait_exponential(multiplier=1, min=4, max=10),
+    reraise=True,
+    stop=stop_after_attempt(3),
+)
+async def send_email(
+    setup: MailgunSetup, arguments: MailgunSendEmailArguments
+) -> MailgunSendEmailOutput:
+    """
+    Sends an email with the provided details using the Mailgun API.
+    """
+
+    try:
+        domain = arguments.from_.split("@")[-1]
+    except Exception as e:
+        msg = f"Error splitting email domain: {e}"
+        raise Exception(msg)
+
+    # Extract domain from from_ address (e.g. "test@example.com" -> "email.example.com")
+    if not domain.startswith("email."):
+        domain = "email." + domain
+
+    # Use API key from env or from setup
+    api_key = mailgun_api_key if setup.api_key == "DEMO_API_KEY" else setup.api_key
+
+    # API URL
+    url = f"https://api.mailgun.net/v3/{domain}/messages"
+
+    # Prepare payload
+    data = {
+        "from": arguments.from_,
+        "to": arguments.to,
+        "subject": arguments.subject,
+        "text": arguments.body,
+    }
+
+    # Add CC and BCC if provided
+    if hasattr(arguments, "cc") and arguments.cc:
```

Comment:
I don't think `hasattr(arguments, "cc")` check is required here

Reply:
yup, removed.